lib/internal/bootstrap_node.js

```diff
@@ -114,6 +114,10 @@
       }
 
       if (process._eval != null && !process._forceRepl) {
+        if (process._syntax_check_only != null) {
+          console.error('--check and --eval flags are mutually exclusive.');
+          process.exit(9);
+        }
```

Comment:
Should these be checked here or should they be checked in `node.cc`? Note that there are other mutually exclusive flags and there is an open PR that does this kind of check directly within `node.cc`